### PR TITLE
perf(playground): code improvements

### LINKED ISSUE #2856: 📎 (rome_playground) Improve playground performance
### Problems:

- Currently formatting (rome & prettier) happens in `App` component in the rendering phase. It seems to result in sluggish keystrokes.
- `react-textarea-code-editor` chokes when I paste 1000 lines of text. Close to 1.2 secs
image

### Possible solutions:

- For the first issue, a simple solution is to debounce formatting. A more complex solution is to move all the computation to  worker thread
- For the second issue, we can use either use codemirror or monaco editors.

### PR DESCRIPTION
**Pull request by vramana**
## Summary

Fixes #2856

- [x] Updated Vite to version 2.9 to add worker support. 
- [x] Updated vite-plugin-svgr to be compatible with Vite 2.9
- [ ] Use `react-codemirror` instead of `react-textarea-code-editor`
- [x] Move formatting from rome and prettier into seperate workers

### CODE REVIEW

File: website/playground/src/types.ts [hunk 1]
```diff
@@ -1,9 +1,31 @@
-import { RomeOutput } from "../pkg";
 import { Dispatch, SetStateAction } from "react";
 
-export enum IndentStyle { Tab = "tab", Space = "space" }
-export enum SourceType { Module = "module", Script = "script" }
-export enum QuoteStyle { Double = "double", Single = "single" }
+export enum IndentStyle {
```
Comment (vramana) [inline website/playground/src/types.ts:3]: There are too many unnecessary code changes. I will fix them a little later.

**Comment by vramana** [comment]
[Screencast from 07-12-2022 12:29:30 PM.webm](https://user-images.githubusercontent.com/3112509/178429149-7ca1293a-8a3e-4bab-8c9c-34c2aa345525.webm)
[Screencast from 07-12-2022 12:30:39 PM.webm](https://user-images.githubusercontent.com/3112509/178429159-0ee81226-2a6a-47a6-a4c4-58b0cba4e0ac.webm)

File: website/playground/package.json [hunk 2]
```diff
@@ -1,41 +1,43 @@
...
+    "preview": "vite preview",
+    "build": "pnpm build:wasm && pnpm build:js",
+    "build:js": "tsc && vite build",
+    "build:wasm": "wasm-pack build --target web --release",
+    "build:wasm-dev": "wasm-pack build --target web --dev",
+    "format": "cargo run -p rome_cli --release -- format --write ./src",
+    "format:rome": "rome format --write src ."
+  },
+  "dependencies": {
+    "@codemirror/lang-javascript": "^6.0.1",
```
[the remark this answers is not in the document]
Comment (vramana) [reply website/playground/package.json:16]: Yes.

**Comment by ematipico** [comment]
I tried to run the app locally, and it seems there's errors in Firefox:

Screenshot 2022-07-14 at 08 55 58

**Comment by vramana** [comment]
image

Apparently firefox is the hold out here.

It doesn't work in dev (Vite restriction) but works in production. https://github.com/vitejs/vite/issues/4586#issuecomment-907079689

**Comment by vramana** [comment]
@ematipico Can you approve workflow so the CI can run?

**Comment by ematipico** [comment]
CI is failing, there's some code not formatted. Just run the command `format:rome` and you should be fine. The last diagnostic is a false positive that we fixed in `main`, so you can ignore it

**Comment by vramana** [comment]
@ematipico Done

**Pull request merged**